ui/src/pages/user_account_page/UserAccountPage.tsx

```diff
@@ -30,85 +33,100 @@ export function UserAccountPage(): JSXElement {
-                </td>
-              </tr>
-            </tbody>
-          </table>
-        </div>
+        <Table>
+          <TableRow
+            label={t('email')}
+            value={user().email}
+            button={<VerifyEmailButton />}
+          />
+          <TableRow
+            label={t('password')}
+            value={'*******'}
+            button={
+              <p class="tooltip tooltip-left" data-tip={t('change_password')}>
+                <button
+                  class={clsx('btn btn-ghost btn-sm')}
+                  onClick={() => openModal('password')}
+                >
+                  <i class="fa-solid fa-edit text-primary" />
+                </button>
+              </p>
+            }
+          />
+          <TableRow
+            label={t('enabled_2fa')}
+            value={
+              <p class="text-lg col-span-2">
+                {user().twoFactorEnabled ? t('yes') : t('no')}
+              </p>
+            }
+            button={
+              <Toggle2FAButton
+                enable2FA={() => openModal('enable2FA')}
+                disable2FA={() => openModal('disable2FA')}
+              />
+            }
+          />
+        </Table>
 
         <div class="mt-4">
           <button
             class="btn btn-error"
-            onClick={() => setOpenDeleteAccountModal(true)}
+            onClick={() => openModal('deleteAcount')}
           >
             <i class="fa-solid fa-trash" />
             {t('delete_account')}
           </button>
         </div>
 
         <ChangePasswordModal
-          isOpen={openPasswordModal()}
-          onClose={() => setOpenPasswordModal(false)}
+          isOpen={modalState().password}
+          onClose={() => closeModal('password')}
         />
 
         <DeleteAccountModal
-          isOpen={openDeleteAccountModal()}
-          onClose={() => setOpenDeleteAccountModal(false)}
+          isOpen={modalState().deleteAcount}
+          onClose={() => closeModal('deleteAcount')}
         />
 
         <Enable2FAModal
-          isOpen={openEnable2FAModal()}
-          onClose={() => setOpenEnable2FAModal(false)}
+          isOpen={modalState().enable2FA}
+          onClose={() => closeModal('enable2FA')}
         />
 
         <Disable2FAModal
-          isOpen={openDisable2FAModal()}
-          onClose={() => setOpenDisable2FAModal(false)}
+          isOpen={modalState().disable2FA}
+          onClose={() => closeModal('disable2FA')}
         />
       </div>
     </>
   )
 }
 
+function TableRow(props: {
+  label: string
+  value: JSXElement | string
+  button?: JSXElement
+}): JSXElement {
+  return (
+    <tr>
+      <td>{props.label}</td>
+      <td>{props.value}</td>
+      <td class="text-end">{props.button}</td>
+    </tr>
+  )
+}
+
+function Table(props: { children: JSXElement[] }): JSXElement {
+  return (
+    <div>
+      <table class="table table-fixed">
+        <tbody>{props.children}</tbody>
+      </table>
+    </div>
+  )
+}
+
```

Comment:
Can you move this into a separate component? Maybe we can also use it in the UsersAdmin?